Sources/Build/Destination.swift

```diff
@@ -0,0 +1,128 @@
+                               extraCCFlags: ["-F", sdkPlatformFrameworksPath.asString],
+                               extraSwiftCFlags: ["-F", platformFrameworkPath,
+                                                  "-F", sdkPlatformFrameworksPath.asString,
+                                                  "-target", "x86_64-apple-macosx10.10",
+                                                  "-sdk", sdkPath],
+                               extraCPPFlags: ["-lc++"])
+        #elseif os(Linux)
+            return Destination(target: "linux-unknown-x86_64",
+                               sdk: AbsolutePath("/"),
+                               toolchain: AbsolutePath(CommandLine.arguments[0],
+                                                       relativeTo: currentWorkingDirectory)
+                                          .parentDirectory.parentDirectory.parentDirectory,
+                               dynamicLibraryExtension: "so",
+                               extraCCFlags: ["-fPIC"],
+                               extraSwiftCFlags: [],
+                               extraCPPFlags: ["-lstdc++"])
+        #else
+            fatalError("unsupported OS, sorry")
+        #endif
+    }
+
+    /// Load a Destination description from a JSON representation from disk.
+    public static func loadFromDescription(path: AbsolutePath,
+                                           fileSystem: FileSystem = localFileSystem) throws -> Destination {
+        let json = try JSON(bytes: try fileSystem.readFileContents(path))
+        return try Destination.init(json: json)
+    }
+
+#if os(macOS)
+    public func sdkPlatformFrameworkPath() -> AbsolutePath? {
+        if let platformPath = try? Process.checkNonZeroExit(args: "xcrun",
+                                                            "--sdk", self.sdk.asString,
+                                                            "--show-sdk-platform-path").chomp()
+                                                            , !platformPath.isEmpty {
+            return AbsolutePath(platformPath).appending(components: "Developer", "Library", "Frameworks")
+        } else {
+            return nil
+        }
+    }
+#endif
+}
+
+extension Destination: JSONMappable {
+    public init(json: JSON) throws {
+        if !((json.get("version")).map ({ $0 == "1" }) ?? false) {
+            throw DestinationError.destinationFileVersionMismatch(problem: "expected version 1")
+        }
+        self.init(target: try json.get("target"),
+                  sdk: AbsolutePath(try json.get("sdk")),
+                  toolchain: AbsolutePath(try json.get("toolchain")),
+                  dynamicLibraryExtension: try json.get("dynamic-library-extension"),
+                  extraCCFlags: try json.get("extra-cc-flags"),
+                  extraSwiftCFlags: try json.get("extra-swiftc-flags"),
+                  extraCPPFlags: try json.get("extra-cpp-flags"))
```

Comment:
my argument is that it makes the syntax confusing but yes it doesn't change anything